docs/filesystems.txt

```diff
@@ -0,0 +1,159 @@
+
+.. list-table::
+   :header-rows: 1
+   :widths: 25 75
+
+   * - Setting
+     - Description
+
+   * - ``driver``
+     - **Required**. Specifies the filesystem driver to use. Must be ``gridfs`` for MongoDB.
+
+   * - ``connection``
+     - The database connection used to store jobs. It must be a ``mongodb`` connection. The driver uses the default connection if a connection is not specified.
+
+   * - ``database``
+     - Name of the MongoDB database for the GridFS bucket. The driver uses the database of the connection if a database is not specified.
+
+   * - ``bucket``
+     - Name or instance of the GridFS bucket. A database can contain multiple buckets identified by their name. Defaults to ``fs``.
+
+   * - ``prefix``
+     - Specifies a prefix for the name of the files that are stored in the bucket. Using a distinct bucket is recommended
+       in order to store the files in a different collection, instead of using a prefix.
+       The prefix should not start with a leading slash ``/``.
+
+   * - ``read-only``
+     - If ``true``, writing to the GridFS bucket is disabled. Write operations will return ``false`` or throw exceptions
+       depending on the configuration of ``throw``. Defaults to ``false``.
+
+   * - ``throw``
+     - If ``true``, exceptions are thrown when an operation cannot be performed. If ``false``,
+	   operations return ``true`` on success and ``false`` on error. Defaults to ``false``.
+
+You can also use a factory or a service name to create an instance of ``MongoDB\GridFS\Bucket``.
+In this case, the options ``connection`` and ``database`` are ignored:
+
+.. code-block:: php
+
+   use Illuminate\Foundation\Application;
+   use MongoDB\GridFS\Bucket;
+
+   'disks' => [
+       'gridfs' => [
+           'driver' => 'gridfs',
+           'bucket' => static function (Application $app): Bucket {
+               return $app['db']->connection('mongodb')
+                   ->getMongoDB()
+                   ->selectGridFSBucket([
+                       'bucketName' => 'avatars',
+                       'chunkSizeBytes' => 261120,
+                   ]);
+           },
+       ],
+   ],
+
+Usage
+-----
+
+A benefit of using Laravel Filesystem is that it provides a common interface
+for all the supported file systems. You can use the ``gridfs`` disk in the same
+way as the ``local`` disk.
+
+.. code-block:: php
+
+   $disk = Storage::disk('gridfs');
+
+   // Write the file "hello.txt" into GridFS
+   $disk->put('hello.txt', 'Hello World!');
+
+   // Read the file
+   echo $disk->get('hello.txt'); // Hello World!
+
+To learn more Laravel File Storage, see
+`Laravel File Storage <https://laravel.com/docs/{+laravel-docs-version+}/filesystem>`__
+in the Laravel documentation.
+
+Versioning
+----------
+
+File names in GridFS are metadata in file documents, which are identified by
+unique ObjectIDs. If multiple documents share the same file name, they are
```

Comment:
S: since this is the first time that file metadata documents are mentioned, I think it would be helpful to describe them 

```suggestion
GridFS creates file documents for each uploaded file. These documents contain metadata, including the file name and a
unique ObjectId. If multiple documents share the same file name, they are
```